static/src/javascripts/bootstraps/enhanced/signup.js

```diff
@@ -0,0 +1,129 @@
+define([
+    'bean',
+    'fastdom',
+    'common/utils/$',
+    'common/utils/fetch',
+    'common/utils/config',
+    'common/modules/identity/api'
+], function (
+    bean,
+    fastdom,
+    $,
+    fetch,
+    config,
+    Id
+) {
+    var classes = {
+      wrapper: 'js-newsletter-meta',
+      signupForm: 'js-email-sub__form',
+      textInput: 'js-newsletter-card__text-input',
+      signupButton: 'js-newsletter-signup-button',
+      styleSignup: 'newsletter-card__lozenge--submit',
+      signupConfirm: 'js-signup-confirmation',
+      previewButton: 'js-newsletter-preview'
+    };
+
+    function hideInputAndShowPreview(el) {
+      fastdom.write(function () {
+          $('.' + classes.textInput, el).addClass('is-hidden');
+          $('.' + classes.signupButton, el).removeClass(classes.styleSignup);
+          $('.' + classes.previewButton, el).removeClass('is-hidden');
+      });
+    }
+
+    function showSignupForm(buttonEl) {
+      var $form = buttonEl.form;
```

Comment:
Last-minute comment on something I just spotted! I don't think `$form` and `$meta` are bonzo wrapped sets, are they? In which case, they shouldn't have the `$` prefix - that convention is to help you know whether you're dealing with a raw DOM element or a bonzo wrapped set. If it's used incorrectly then it's harmful rather than useful - someone may waste time and get frustrated trying to call bonzo methods on `$form` before realising it's actually just a DOM element